index.d.ts

```diff
@@ -68,57 +96,52 @@ declare module "hyperapp" {
   // ---------------------------------------------------------------------------
 
   // A dispatched action handles an event in the context of the current state.
-  type Dispatch<S> = (action: Action<S>, payload?: Payload<any>) => void
+  type Dispatch<S> = (action: Action<S>, payload?: any) => void
+
+  // A dispatchable entity, when processed, causes a state transition.
+  type Dispatchable<S> = S | StateWithEffects<S> | Action<S>
 
   // An action transforms existing state and/or wraps another action.
   type Action<S, P = any> = ActionTransform<S, P> | ActionWithPayload<S, P>
-  type ActionTransform<S, P = any> = (
-    state: State<S>,
-    payload: Payload<P>
-  ) => State<S> | StateWithEffects<S> | Action<S>
-  type ActionWithPayload<S, P> = [ActionTransform<S, P>, Payload<P>]
+  type ActionTransform<S, P = any> = (state: S, payload: P) => Dispatchable<S>
```

Comment:
There's something redundant in ActionsTransform and Dispatchable. `Dispatchable` could very well be an `(state: S, payload: P)` too. I think there's room for simplification here.

Reply:
Well, `type Dispatchable<S> = S | StateWithEffects<S> | Action<S>`.  So, it can simplify into `(state: S, payload: P) => Dispatchable<S>` but of course there's more going on.

If there's a way to simplify even further that would be great but I don't see it.

Reply:
@jorgebucaran Mind if I resolve this one? I don't think a change can be made here without introducing redundancy.

Reply:
Let's give @zaceno a chance to review the PR too. 👌

Reply:
I see no problem with the definition of `ActionTransform`. 

_Technically_  I think `Dispatch` should probably be:

```ts
type Dispatch<S> = (action: Dispatchable<S>,  payload?: any) => void
```

because it's fully possible to `dispatch({foo: 'bar'})`, or `dispatch([{foo: 'bar'}, http('example.com')])`

But if we want to discourage that useage then the current definition is fine (I don't feel strongly about this – just pointing it out)

Reply:
Fascinating, I had not even considered using dispatch like that. I'm not sure if there's a use case for not using an action with dispatch but I'm open to that possibility. So, I wouldn't be opposed to expanding the definition for `Dispatch` in this way.

Reply:
Well, the most obvious use case is `init`. It's usually not an action, but just the initial state or a state+effect tuple.

Reply:
The most obvious use case for that would be `init`, usually not an action, but just the state or a state+effect tuple.

Reply:
Oh, yeah, haha !

Reply:
I updated the PR to use an expanded definition of `Dispatch` similar  to the one suggested by @zaceno.